specs/0118-notification-center.md

```diff
@@ -0,0 +1,119 @@
+          - Tooltip on `Maximum`: `Leave empty to include all numbers greater than or equal to Minimum.`
+        - `Timeframe filter`
+          - Checkbox (default empty): `Enable timeframe filter`
+            - Checking the box displays datepickers
+          - Datepicker: `From:` (Default value: today's date)
+          - Datepicker: `To:` (Default value: today's date)
+      - Inputs for `Last login`
+        - `Timeframe filter`
+          - Datepicker: `From:` (Default value: today's date)
+          - Datepicker: `To:` (Default value: today's date)
+- Header: `Transactional email`
+  - By default, emails are sent to users who have enabled opt-in consent. (Does not apply to in-app notifications.)
+  - This “Transactional” option flags emails as transactional and sends emails to users regardless of their opt-in status. However, this email content would be restricted to content transactional in nature (e.g. account status notifications, etc.).
+  - Checkbox (default empty): `This is a transactional email`
+  - Text description: `Transactional emails are sent to users who have not opted in to email communication. Transactional emails are restricted in the type of content that can be included.`
+  - Validation:
+    - Error if `Transactional email` enabled but `Message type` does not include `Email`: `You cannnot enable the transactional email option if the Message type does not include email.`
+- Button: `Review message`
+  - A send button is not displayed on the same page as the message editor and recipient selection to avoid sending an incomplete message or a message to the wrong recipients.
+  - When pressed a validation step is performed on inputs to ensure there are no issues with the form. If there are no issues, the user proceeds to the Review/Confirmation page.
+
+- Title: `Review message`
+- A review step is required to reduce errors because messages could be sent to a large number of people. This page allows you to review all aspects of the email before sending, including:
+ - A rendered version of the message to ensure things like Markdown tags or html don’t have issues.
+ - A summary of who will be receiving the nessage:
+    - The number of recipients
+    - The filters being applied
+  - Whether this is a transactional email. If enabled, a warning will be shown: `Warning: transactional emails are sent to users who have not opted in to email communication. Transactional emails are restricted in the type of content that can be included. When in doubt, please review with legal.`
+  - Button: `Edit`
+    - Clicking the `Edit` button or navigating back in the browser brings you to the previous page
+    - The previously entered content is preserved when `Edit` or the browser's back button is useds so work is not lost
+  - Checkbox (default empty): `I’ve reviewed the work and this is ready to be sent.`
+    - This is another safeguard to avoid accidentally sending messages without thorough review.
```

Comment:
I like the idea of preventing sending messages accidentally. 

A checkbox in addition to the Review step seems quite involved, though.

Reply:
See my answer above.

Reply:
Every single one of us has made mistakes before, especially with notifications. I prefer to err on the side of caution.

Reply:
Can you share an example of such UX pattern from a popular application? Where you are already on the Review step, and you need to click a ("This is ready to be sent") checkbox before clicking Send?

I don't remember anything like that from my email clients, web shops or online banks (which, TBF, usually require an extra step on another device, but that's due to authentication, not sanity check).

Not even professional marketing service [MailerLite](https://www.mailerlite.com/) uses that (I didn't check others). And we're building just an MVP of a tool like MailerLite.

I 100% understand we'd like to prevent mistakes, but I don't see how this checkbox helps with that. I'm happy to be convinced otherwise with good examples. 😄

Reply:
I've removed this check for now. We could argue either way but we have at least 2 clicks before an email gets sent which is at least the bare minimum.